src/mobdebug.lua

```diff
@@ -135,6 +136,13 @@ local c, d = "Paul Kulchenko", "Lua serializer and pretty printer"
 local snum = {[tostring(1/0)]='1/0 --[[math.huge]]',[tostring(-1/0)]='-1/0 --[[-math.huge]]',[tostring(0/0)]='0/0'}
 local badtype = {thread = true, userdata = true, cdata = true}
 local getmetatable = debug and debug.getmetatable or getmetatable
+local oldtostring = tostring
+local function tostring( x )
+  local mt = getmetatable( x )
+  local ret = oldtostring( c )
```

Comment:
Aaactually, it would error if `x` is not a table. So yep, a type check is necessary.